app/src/main/java/com/github/se/eduverse/ui/profile/ProfileScreen.kt

```diff
@@ -208,53 +311,29 @@ private fun PublicationsGrid(
         }
   } else {
     LazyVerticalGrid(
-        columns = GridCells.Fixed(2),
+        columns = GridCells.Fixed(3),
         modifier = modifier.fillMaxSize(),
-        contentPadding = PaddingValues(16.dp),
-        horizontalArrangement = Arrangement.spacedBy(8.dp),
-        verticalArrangement = Arrangement.spacedBy(8.dp)) {
+        contentPadding = PaddingValues(1.dp),
+        horizontalArrangement = Arrangement.spacedBy(1.dp),
+        verticalArrangement = Arrangement.spacedBy(1.dp)) {
           items(publications) { publication ->
             PublicationItem(
                 publication = publication,
-                onClick = { onPublicationClick(publication) },
+                onClick = { selectedPublication = publication },
                 modifier = Modifier.testTag("publication_item_${publication.id}"))
           }
         }
+
+    // Show detail dialog when a publication is selected
+    selectedPublication?.let { publication ->
+      PublicationDetailDialog(publication = publication, onDismiss = { selectedPublication = null })
+    }
   }
 }
 
```

Comment:
The code demonstrates a strong modular structure, with distinct functions such as PublicationItem and PublicationDetailDialog, making it easy to maintain and extend. The use of testTag for various UI elements reflects a good practice for UI testing, facilitating the verification process. Additionally, the handling of different media types (PHOTO and VIDEO) is well-executed, ensuring appropriate display for each type. The implementation of visual elements like AspectRatio and RoundedCornerShape adds a pleasant aesthetic to the user interface, enhancing the overall user experience.